test/quantum_objects.jl

```diff
@@ -51,19 +51,22 @@
 
     # unsupported type of dims
     @testset "unsupported dims" begin
-        @test_throws ArgumentError Qobj(rand(2, 2), dims = 2)
-        @test_throws ArgumentError Qobj(rand(2, 2), dims = [2.0])
-        @test_throws ArgumentError Qobj(rand(2, 2), dims = [2.0 + 0.0im])
-        @test_throws DomainError Qobj(rand(2, 2), dims = [0])
-        @test_throws DomainError Qobj(rand(2, 2), dims = [2, -2])
+        @test_throws ArgumentError Qobj(rand(2, 2), dims = (2.0,))
+        @test_throws ArgumentError Qobj(rand(2, 2), dims = (2.0 + 0.0im,))
+        @test_throws DomainError Qobj(rand(2, 2), dims = (0,))
```

Comment:
maybe just put `dims = 2.0`, `dims = 2.0 + 0.0im`, and `dims = 0` here?
since we already defined `_check_dims(dims::Any)` and `_check_dims(dims::Integer)`